File: backend/VendeTuAutoYa.Api/Models/UserProfile.cs
using System.ComponentModel.DataAnnotations;

namespace VendeTuAutoYa.Api.Models
{
    public class UserProfile
    {
        public int Id { get; set; }
        
        // Campos comunes para todos los usuarios
        [MaxLength(100)]
        public string? FirstName { get; set; }
        
        [MaxLength(100)]
        public string? LastName { get; set; }
        
        [MaxLength(20)]
        public string? Phone { get; set; }
        
        [MaxLength(20)]
        public string? DocumentNumber { get; set; } // DNI/CUIT/CUIL
        
        // Campos específicos para concesionarios
        [MaxLength(11)]
        public string? Cuit { get; set; }
        
        [MaxLength(200)]
        public string? BusinessName { get; set; }
        
        [MaxLength(200)]
        public string? LegalRepresentative { get; set; }
        
        // Campos de verificación
        public bool IsProfileComplete { get; set; } = false;
        public bool IsDocumentationComplete { get; set; } = false;
        public bool IsAddressComplete { get; set; } = false;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        // Navigation properties
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public Address? Address { get; set; }
        public ICollection<Document> Documents { get; set; } = new List<Document>();
    }
}

File: frontend/src/types/userProfile.ts
// Tipos para el perfil de usuario
export interface Address {
  id?: number
  street: string
  number: string
  floor?: string
  apartment?: string
  city: string
  province: string
  postalCode?: string
}

export interface Document {
  id: number
  documentType: string
  fileName: string
  contentType: string
  fileSize: number
  uploadedAt: string
  isActive: boolean
}

export interface UserProfile {
  id?: number
  firstName?: string
  lastName?: string
  phone?: string
  documentNumber?: string
  cuit?: string
  businessName?: string
  legalRepresentative?: string
  isProfileComplete: boolean
  isDocumentationComplete: boolean
  isAddressComplete: boolean
  address?: Address
  documents: Document[]
  createdAt?: string
  updatedAt?: string
}

// DTOs para requests
export interface UpdateUserProfileRequest {
  firstName?: string
  lastName?: string
  phone?: string
  documentNumber?: string
  cuit?: string
  businessName?: string
  legalRepresentative?: string
  address?: {
    street: string
    number: string
    floor?: string
    apartment?: string
    city: string
    province: string
    postalCode?: string
  }
}

// Response para el estado de completitud
export interface CompletionStatus {
  isProfileComplete: boolean
  isAddressComplete: boolean
  isDocumentationComplete: boolean
  overallComplete: boolean
  completionPercentage: number
}

// Tipos de documentos permitidos
export type DocumentType = 'DNI' | 'Estatuto' | 'AFIP'

// Response para upload de documentos
export interface DocumentUploadResponse {
  id: number
  documentType: string
  fileName: string
  contentType: string
  fileSize: number
  uploadedAt: string
  isActive: boolean
}